pkg/actuators/machine/actuator.go

```diff
@@ -233,8 +240,55 @@ func (a *Actuator) Delete(context context.Context, cluster *clusterv1.Cluster, m
 	return nil
 }
 
+type glogLogger struct{}
+
+func (gl *glogLogger) Log(v ...interface{}) {
+	glog.Info(v...)
+}
+
+func (gl *glogLogger) Logf(format string, v ...interface{}) {
+	glog.Infof(format, v...)
+}
+
 // DeleteMachine deletes an AWS instance
 func (a *Actuator) DeleteMachine(cluster *clusterv1.Cluster, machine *clusterv1.Machine) error {
+	// Drain node before deleting
+	if machine.ObjectMeta.Annotations["drain-node"] == "True" && machine.Status.NodeRef != nil {
```

Comment:
we should scope this similar to `cluster.k8s.io/machine`

Reply:
As long as the `drain-node` label is not upstream accepted, I would not issue `cluster.k8s.io` prefix. Plus, the label will get removed anyway. Though, we can do `openshift.io/drain-node` for now. Hmm?